core/api.go

```diff
@@ -265,11 +267,7 @@ func (a *API) authzHandler(mux *http.ServeMux, handler http.Handler) http.Handle
 	auth := authz.NewAuthorizer(a.raftDB, grantPrefix, policyByRoute)
 	auth.GrantInternal(a.internalSubj)
 	return http.HandlerFunc(func(rw http.ResponseWriter, req *http.Request) {
-		// return failure early if this path isn't legit
-		if _, pat := mux.Handler(req); pat != req.URL.Path {
-			errorFormatter.Write(req.Context(), rw, errNotFound)
-			return
-		}
+		// TODO(tessr): check that this path exists; return early if this path isn't legit
```

Comment:
This is kind of a bummer, but getting all the pattern matching right here, too, on all the routes (which includes routes which haven't been added to this mux yet) turned out to be such a pain in the butt.